File: flutter_project/lib/screens/settings_screen.dart
/// Settings Screen - 配置屏幕
/// 功能：配置API Key、Base URL、总结偏好和并行模式
library settings_screen;

import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../theme/design_system.dart';
import '../../models/models.dart';
import '../../providers/app_state_provider.dart';

class SettingsScreen extends StatefulWidget {
  const SettingsScreen({Key? key}) : super(key: key);

  @override
  State<SettingsScreen> createState() => _SettingsScreenState();
}

class _SettingsScreenState extends State<SettingsScreen> {
  late TextEditingController _apiKeyController;
  late TextEditingController _baseUrlController;
  late TextEditingController _userPromptController;

  @override
  void initState() {
    super.initState();
    final appState = context.read<AppStateProvider>();
    _apiKeyController = TextEditingController(text: appState.config.apiKey);
    _baseUrlController = TextEditingController(text: appState.config.baseUrl);
    _userPromptController = TextEditingController(text: appState.config.userPrompt);
  }

  @override
  void dispose() {
    _apiKeyController.dispose();
    _baseUrlController.dispose();
    _userPromptController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      body: Consumer<AppStateProvider>(
        builder: (context, appState, _) => SafeArea(
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(AppSpacing.screenPaddingHorizontal),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Header
                SizedBox(
                  height: AppSpacing.lg,
                ),
                Text(
                  'Video Summarizer',
                  style: AppFonts.title.copyWith(
                    color: AppColors.textPrimary,
                  ),
                ),
                const SizedBox(height: 4),
                Text(
                  '多模态智能总结',
                  style: AppFonts.subtitle.copyWith(
                    color: AppColors.textSecondary,
                  ),
                ),
                const SizedBox(height: AppSpacing.xl),

                // API Configuration Section
                Text(
                  'API 配置',
                  style: AppFonts.label.copyWith(
                    color: AppColors.textPrimary,
                  ),
                ),
                const SizedBox(height: AppSpacing.md),

                // API Key Input
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      'OpenAI API Key',
                      style: AppFonts.caption.copyWith(
                        color: AppColors.textSecondary,
                      ),
                    ),
                    const SizedBox(height: 4),
                    TextField(
                      controller: _apiKeyController,
                      obscureText: true,
                      decoration: InputDecoration(
                        hintText: 'sk-...',
                        hintStyle: AppFonts.body.copyWith(
                          color: AppColors.textTertiary,
                        ),
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: AppSpacing.lg),

                // Base URL Input
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      'Base URL',
                      style: AppFonts.caption.copyWith(
                        color: AppColors.textSecondary,
                      ),
                    ),
                    const SizedBox(height: 4),
                    TextField(
                      controller: _baseUrlController,
                      decoration: InputDecoration(
                        hintText: 'https://api.openai.com/v1',
                        hintStyle: AppFonts.body.copyWith(
                          color: AppColors.textTertiary,
                        ),
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: AppSpacing.xl),

                // Summary Preference Section
                Text(
                  '总结偏好',
                  style: AppFonts.label.copyWith(
                    color: AppColors.textPrimary,
                  ),
                ),
                const SizedBox(height: AppSpacing.md),

                TextField(
                  controller: _userPromptController,
                  maxLines: 4,
                  decoration: InputDecoration(
                    hintText: '请输入总结重点（可选）',
                    hintStyle: AppFonts.body.copyWith(
                      color: AppColors.textTertiary,
                    ),
                  ),
                ),
                const SizedBox(height: AppSpacing.xl),

                // Concurrency Mode Section
                Text(
                  '并行模式',
                  style: AppFonts.label.copyWith(
                    color: AppColors.textPrimary,
                  ),
                ),
                const SizedBox(height: AppSpacing.md),

                // ThreadPool Option
                _ModeOption(
                  label: '✓ ThreadPool (默认)',
                  isSelected: appState.config.concurrencyMode == ConcurrencyMode.threadPool,
                  onTap: () async {
                    await appState.setConcurrencyMode(ConcurrencyMode.threadPool);
                  },
                ),
                const SizedBox(height: AppSpacing.md),

                // Send API Option
                _ModeOption(
                  label: '○ Send API',
                  isSelected: appState.config.concurrencyMode == ConcurrencyMode.sendAPI,
                  onTap: () async {
                    await appState.setConcurrencyMode(ConcurrencyMode.sendAPI);
                  },
                ),
                const SizedBox(height: AppSpacing.xl),

                // Save Button
                SizedBox(
                  width: double.infinity,
                  height: 44,
                  child: ElevatedButton(
                    style: ElevatedButton.styleFrom(
                      backgroundColor: AppColors.primary,
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(AppBorderRadius.medium),
                      ),
                    ),
                    onPressed: () async {
                      await appState.saveApiConfig(
                        apiKey: _apiKeyController.text,
                        baseUrl: _baseUrlController.text,
                      );
                      await appState.saveUserPrompt(_userPromptController.text);
                      
                      if (mounted) {
                        ScaffoldMessenger.of(context).showSnackBar(
                          const SnackBar(
                            content: Text('配置已保存'),
                            duration: Duration(seconds: 2),
                          ),
                        );
                      }
                    },
                    child: Text(
                      '保存配置',
                      style: AppFonts.label.copyWith(
                        color: AppColors.white,
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: AppSpacing.xl),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

class _ModeOption extends StatelessWidget {
  final String label;
  final bool isSelected;
  final VoidCallback onTap;

  const _ModeOption({
    required this.label,
    required this.isSelected,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        padding: const EdgeInsets.all(AppSpacing.md),
        decoration: BoxDecoration(
          border: Border.all(
            color: isSelected ? AppColors.primary : AppColors.border,
            width: isSelected ? 2 : 1,
          ),
          borderRadius: BorderRadius.circular(AppBorderRadius.medium),
          color: isSelected ? AppColors.highlightBg : AppColors.white,
        ),
        child: Text(
          label,
          style: AppFonts.body.copyWith(
            color: isSelected ? AppColors.primary : AppColors.textTertiary,
          ),
        ),
      ),
    );
  }
}

File: pencil/screens/SettingsScreen.js
/**
 * SettingsScreen.js - 配置屏幕
 * 功能：配置API Key、Base URL、总结偏好和并行模式
 */

import React, { useState } from 'react';
import {
  View,
  ScrollView,
  StyleSheet,
  SafeAreaView,
  Text,
  TextInput,
  TouchableOpacity,
  StatusBar,
  Platform,
} from 'react-native';
import {
  COLORS,
  FONTS,
  SPACING,
  BORDER_RADIUS,
} from '../App';

export default function SettingsScreen() {
  const [apiKey, setApiKey] = useState('sk-...');
  const [baseUrl, setBaseUrl] = useState('https://api.openai.com/v1');
  const [userPrompt, setUserPrompt] = useState('');
  const [selectedMode, setSelectedMode] = useState('threadpool');

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar
        barStyle="light-content"
        backgroundColor="#000000"
        translucent={false}
      />
      
      {/* Status Bar */}
      <View style={styles.statusBar}>
        <Text style={styles.statusTime}>9:41</Text>
        <Text style={styles.statusIcon}>📶</Text>
      </View>

      <ScrollView
        style={styles.content}
        contentContainerStyle={styles.contentContainer}
        showsVerticalScrollIndicator={false}
      >
        {/* Header */}
        <View style={styles.header}>
          <Text style={[FONTS.title, { color: COLORS.textPrimary }]}>
            Video Summarizer
          </Text>
          <Text style={[FONTS.subtitle, { color: COLORS.textSecondary, marginTop: SPACING.xs }]}>
            多模态智能总结
          </Text>
        </View>

        {/* API Configuration Section */}
        <View style={styles.section}>
          <Text style={[FONTS.label, { color: COLORS.textPrimary, marginBottom: SPACING.md }]}>
            API 配置
          </Text>

          {/* API Key Input */}
          <View style={styles.fieldGroup}>
            <Text style={[FONTS.caption, { color: COLORS.textSecondary, marginBottom: SPACING.xs }]}>
              OpenAI API Key
            </Text>
            <TextInput
              style={styles.input}
              placeholder="sk-..."
              value={apiKey}
              onChangeText={setApiKey}
              secureTextEntry={true}
              placeholderTextColor={COLORS.textTertiary}
            />
          </View>

          {/* Base URL Input */}
          <View style={styles.fieldGroup}>
            <Text style={[FONTS.caption, { color: COLORS.textSecondary, marginBottom: SPACING.xs }]}>
              Base URL
            </Text>
            <TextInput
              style={styles.input}
              placeholder="https://api.openai.com/v1"
              value={baseUrl}
              onChangeText={setBaseUrl}
              placeholderTextColor={COLORS.textTertiary}
            />
          </View>
        </View>

        {/* Summary Preference Section */}
        <View style={styles.section}>
          <Text style={[FONTS.label, { color: COLORS.textPrimary, marginBottom: SPACING.md }]}>
            总结偏好
          </Text>

          <TextInput
            style={[styles.textArea]}
            placeholder="请输入总结重点（可选）"
            value={userPrompt}
            onChangeText={setUserPrompt}
            multiline
            numberOfLines={4}
            placeholderTextColor={COLORS.textTertiary}
            textAlignVertical="top"
          />
        </View>

        {/* Concurrency Mode Section */}
        <View style={styles.section}>
          <Text style={[FONTS.label, { color: COLORS.textPrimary, marginBottom: SPACING.md }]}>
            并行模式
          </Text>

          {/* ThreadPool Option */}
          <TouchableOpacity
            style={[
              styles.modeOption,
              selectedMode === 'threadpool' && styles.modeOptionActive,
            ]}
            onPress={() => setSelectedMode('threadpool')}
          >
            <Text style={[
              FONTS.body,
              {
                color: selectedMode === 'threadpool' ? COLORS.primary : COLORS.textTertiary,
              }
            ]}>
              ✓ ThreadPool (默认)
            </Text>
          </TouchableOpacity>

          {/* Send API Option */}
          <TouchableOpacity
            style={[
              styles.modeOption,
              { marginTop: SPACING.md },
              selectedMode === 'send_api' && styles.modeOptionActive,
            ]}
            onPress={() => setSelectedMode('send_api')}
          >
            <Text style={[
              FONTS.body,
              {
                color: selectedMode === 'send_api' ? COLORS.primary : COLORS.textTertiary,
              }
            ]}>
              Send API (试点)
            </Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.background,
  },
  statusBar: {
    height: 62,
    backgroundColor: '#000000',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: SPACING.lg,
  },
  statusTime: {
    fontSize: 14,
    fontWeight: '600',
    color: COLORS.white,
  },
  statusIcon: {
    fontSize: 11,
    color: COLORS.white,
  },
  content: {
    flex: 1,
  },
  contentContainer: {
    padding: SPACING.lg,
    paddingBottom: 100,
  },
  header: {
    marginBottom: SPACING.xxl,
  },
  section: {
    marginBottom: SPACING.xxl,
  },
  fieldGroup: {
    marginBottom: SPACING.lg,
  },
  input: {
    backgroundColor: COLORS.white,
    borderRadius: BORDER_RADIUS.medium,
    borderWidth: 1,
    borderColor: COLORS.border,
    paddingHorizontal: SPACING.lg,
    paddingVertical: SPACING.md,
    fontSize: 12,
    color: COLORS.textPrimary,
  },
  textArea: {
    backgroundColor: COLORS.white,
    borderRadius: BORDER_RADIUS.medium,
    borderWidth: 1,
    borderColor: COLORS.border,
    padding: SPACING.lg,
    fontSize: 12,
    color: COLORS.textPrimary,
    minHeight: 80,
  },
  modeOption: {
    backgroundColor: COLORS.white,
    borderRadius: BORDER_RADIUS.medium,
    borderWidth: 1,
    borderColor: COLORS.border,
    paddingHorizontal: SPACING.lg,
    paddingVertical: SPACING.md,
  },
  modeOptionActive: {
    borderColor: COLORS.primary,
    borderWidth: 2,
  },
});
